internal/session/session.go

```diff
@@ -50,53 +43,23 @@ func (s *DefaultSession) Logout() error {
 	return s.connection.LogoutSession(s.token)
 }
 
-func (s *DefaultSession) RequestBody(url, version string, payload interface{}) ([]byte, string, error) {
-	if s.popRestricted {
-		if s.key == nil {
-			return nil, "", errors.New("request requires a signed body, but no signing key was configured")
-		}
-		requestBody, err := signedJWTBody(s, url, version, payload)
-		if err != nil {
-			return nil, "", err
-		}
-		return requestBody, string(client.ApplicationJOSE), nil
-	} else if payload == nil {
-		return []byte(""), string(client.ApplicationJSON), nil
-	} else {
-		b, err := json.Marshal(payload)
-		if err != nil {
-			return nil, "", err
-		}
-		return b, string(client.ApplicationJSON), nil
-	}
+type PoPSession struct {
+	*DefaultSession
```

Comment:
It would be safer to embed `DefaultSession` instead of `*DefaultSession`. `*PoPSession` would still satisfy the connection interface and we remove a potential source of nil pointer exceptions.